driver/vm.go

```diff
@@ -490,26 +498,52 @@ func addDisk(_ *Driver, devices object.VirtualDeviceList, config *CreateConfig)
 		return nil, err
 	}
 
-	disk := &types.VirtualDisk{
-		VirtualDevice: types.VirtualDevice{
-			Key: devices.NewKey(),
-			Backing: &types.VirtualDiskFlatVer2BackingInfo{
-				DiskMode:        string(types.VirtualDiskModePersistent),
-				ThinProvisioned: types.NewBool(config.DiskThinProvisioned),
+	for _, dc := range config.Storage {
+		if dc.DiskName != "" {
+			dc.DiskName += ".vmdk"
+		}
+
+		if dc.DiskType == "" && config.GlobalDiskType != "" {
+			dc.DiskType = config.GlobalDiskType
+		}
+
+		if dc.DiskType == "thin" {
```

Comment:
There should be input (`DiskType`) validation in some other place. 
Probably `DiskEagerlyScrub` and `DiskThinProvisioned` cloud be package-private (lowerCase)

Reply:
Thanks, I will look into this.